source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md

```diff
@@ -0,0 +1,147 @@
+* You'll need to bind mount an area of the file system, so that `buildah bud` can build up and then commit the image to the host file system. This can be `/var/lib/containers` if you like, so as it’s immediately shared shared on the hose, or it can be some build sandbox area etc. You also need to bind mount the Dockerfile path so that Buildah can see the Dockerfile. So for the example I’ll use the bind mount parameter `-v` or `--volume` parameter for `podman run`.
+
+Use your favorite Nginx Dockerfile or the one below. 
+
+```
+# Base on the Fedora
+FROM fedora:latest
+MAINTAINER http://fedoraproject.org/wiki/Cloud
+
+RUN dnf -y update && dnf clean all
+RUN dnf -y install nginx && dnf clean all
+RUN echo "daemon off;" >> /etc/nginx/nginx.conf
+RUN echo "nginx on Fedora" > /usr/share/nginx/html/index.html
+
+EXPOSE 80
+
+ENTRYPOINT [ "/usr/sbin/nginx" ]
+```
+
+The following command builds an Nginx container image called ‘nginx’. Obviously using Podman (or Docker) is overkill because they both have build capabilities. Ideally this container would be run in a pod on kubernetes through CRI-O. My Dockerfile is in my `/home/whenry/dfs` directory (shorter for dockerfiles). So I bind mount that directory to `/tmp` inside the container. Inside it’s `/tmp/Dockerfile`.
+
+```
+# podman run --privileged --network host -v /var/lib/containers:/var/lib/containers:rw  -v /home/whenry/dfs:/tmp:Z buildah bud -f /tmp/Dockerfile -t nginx 
+```
+
+Currently, due to OverlayFS requiring root privileges, this container must run in privileged mode. This requirement should be deprecated when Fedora’s OverlayFS does not require a privileged user.
+
+## Current issues with non privileged mode
+
+There is a requirement in large clusters for buildah not to run in privileged mode. There are ways to mitigate this but there will be cases where people demand it.
+
+I had prepared an example of how to do this but I will keep it for a later blog.  Here are some of the current constraints I encountered trying to solve the non-privileged problem
+
+* If you want to share the builds immediately through the hosts `/var/lib/containers` then you need to run Buildah as privileged. You could build somewhere else and then move the images to `/var/lib/containers` - see below. 
+* If you don't care to share on the host immediately then you can use unprivileged (don’t use --privileged) but you need to use a different directory than /var/lib/containers. However currently OverlayFS requires privileged mode on Fedora/RHEL. This should be fixed soon. You could use `--storage-driver=vfs`.
+* If you decide to build inside the container you can do that unpriv but you hit the overlay on overlay issue and then need to run with storage-driver vfs. This worked at one point but I’ve seen a regress.
```

Comment:
Perhaps: "unpriv but you hit the overlay on overlay issue and then need to run with storage-driver vfs" to "unpriv but you will likely hit the overlay on overlay issue and will then need to run with storage-driver vfs"

Reply:
+1 changed to 
unprivileged, but you hit the overlay on overlay issue and will then need to run with `--storage-driver vfs`

Reply:
:+1: